### ISSUE TITLE
vivaldi could not find the Qt platform plugin "xcb" in ""

### ISSUE THREAD

**Issue by SyedAhkam** [NixOS/nixpkgs#309056]
### Describe the bug

When trying to run vivaldi  6.6.3271.61 stable on the nixos 'unstable' channel, I get the following error: 

```
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

[0504/204138.752892:ERROR:elf_dynamic_array_reader.h(64)] tag not found
[0504/204138.753036:ERROR:elf_dynamic_array_reader.h(64)] tag not found
[0504/204138.753600:ERROR:process_memory_range.cc(75)] read out of range
fish: Job 1, 'vivaldi' terminated by signal SIGABRT (Abort)

```
This happens on both X & Wayland.

### Steps To Reproduce

1. Install vivaldi by adding inside `configuration.nix`
2. Run it on the command line
3. Witness error

### Expected behavior

Expect the application to run normally & not exit as before launching.

### Screenshots

N/A

### Additional context

Similar issue in another program: https://github.com/NixOS/nixpkgs/issues/301819

### Notify maintainers

@otwieracz @badmutex

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.29, NixOS, 24.05 (Uakari), 24.05.20240502.63c3a29`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/p69bcs7ma6ijj8v9xsrg3nq3nn8ryn95-source`
```

**Comment by matklad** [comment]
See https://github.com/NixOS/nixpkgs/pull/292148

**Comment by SyedAhkam** [comment]
Thanks for the response, I wasn't aware of that issue and had no idea how deep this problem really is.

I was almost about to send a PR adding in `qt5.wrapQtAppsHook` but now I guess I shouldn't. Although I'm happy I got it running locally with the wrapping.

**Comment by Lippiece** [comment]
> Thanks for the response, I wasn't aware of that issue and had no idea how deep this problem really is.
> 
> I was almost about to send a PR adding in `qt5.wrapQtAppsHook` but now I guess I shouldn't. Although I'm happy I got it running locally with the wrapping.

Can you please elaborate? I'm having this issue now.

**Comment by SyedAhkam** [comment]
@Lippiece 

yes, what you can do is wrap it by overriding the attrs when putting vivaldi in your config:

```nix
# Vivaldi with forceful qt app wrapping
  (vivaldi.overrideAttrs (oldAttrs: {
    dontWrapQtApps = false;
    dontPatchELF = true;
    nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ pkgs.kdePackages.wrapQtAppsHook ];
  }))
 ```

**Comment by DavKBe** [comment]
@SyedAhkam 

your solution didn't work for me.

qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

[0110/004139.992484:ERROR:elf_dynamic_array_reader.h(64)] tag not found
[0110/004139.992913:ERROR:process_memory_range.cc(75)] read out of range
Aborted (core dumped)

**Comment by SyedAhkam** [comment]
@DavKBe please look into #292148 and report there

**Comment by wineee** [comment]
> Thanks for the response, I wasn't aware of that issue and had no idea how deep this problem really is.
> 
> I was almost about to send a PR adding in `qt5.wrapQtAppsHook` but now I guess I shouldn't. Although I'm happy I got it running locally with the wrapping.

Using wrapQtAppsHook should be the right solution https://github.com/NixOS/nixpkgs/pull/406915
